### PR TITLE
Basic Mob Gorillas

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Now we can make basic mobs with hands easily so I did, they don't actually use their hands for anything with AI.
In the future we can come back and share the monkey AI where they pick up items to hit people with, but frankly few weapons are more deadly than a gorilla's fists.

IIRC I didn't really change their behaviour much, this is mostly just a straight conversion. Main difference is that they will prioritise eating nearby bananas and fruit salads over punching people.

When I make these conversions nowadays I need to decide between "does this attack at the speed that it did as an NPC mob or the speed it did as a player?"
I am arbitrarily deciding that gorillas are usually not players and electing for the former, but tell me if you disagree.

I also made "show basic inhand sprites" into a component shared by Gorillas, Drones, and Dextrous Guardians (all also now available to become basic, once I get around to it),

And I added an AI behaviour to run a basic emote. This is similar but different to "random speech", which kind of sucks and needs rewriting anyway.
Gorillas don't speak, only ooga.

## Why It's Good For The Game

https://www.youtube.com/watch?v=npuuTBlEb1U

## Changelog

:cl:
refactor: Gorillas now use the basic mob framework. Please report any unusual side effects.
/:cl:

### CODE REVIEW

File: code/modules/spells/spell_types/conjure/simian.dm [hunk 1]
```diff
@@ -14,14 +14,14 @@
 	invocation_type = INVOCATION_SHOUT
 
 	summon_radius = 2
-	summon_type = list(/mob/living/carbon/human/species/monkey/angry, /mob/living/carbon/human/species/monkey/angry, /mob/living/simple_animal/hostile/gorilla/lesser)
+	summon_type = list(/mob/living/carbon/human/species/monkey/angry, /mob/living/carbon/human/species/monkey/angry, /mob/living/basic/gorilla/lesser)
```
Comment (san7890) [inline code/modules/spells/spell_types/conjure/simian.dm:17]: multiline and alphabetize? it's long enough

File: code/modules/mob/inventory.dm [hunk 2]
```diff
@@ -286,6 +286,12 @@
 	I.dropped(src)
 	return FALSE
 
+/// Returns true if a mob is holding something
+/mob/proc/is_holding_items()
+	for(var/obj/item/held in held_items)
```
Comment (san7890) [inline code/modules/mob/inventory.dm:291]:
```suggestion
	if(locate(/obj/item) in held_items)
```

idk if this is the right syntax but i think it's better than a forloop here imo

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/datums/ai/basic_mobs/basic_subtrees/run_emote.dm [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/// Intermittently run an emote
+/datum/ai_planning_subtree/run_emote
+	var/emote_key = BB_EMOTE_KEY
```
Comment (mc-oofert) [inline code/datums/ai/basic_mobs/basic_subtrees/run_emote.dm:3]: didnt we have this kinda thing already for clowns

Comment (Jacquerel) [reply code/datums/ai/basic_mobs/basic_subtrees/run_emote.dm:3]: no

**Pull request merged**